src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
-   * This sets the Closer interface for this client.  
-   * Once set the client will call .onClose on it once it a socket close is detected. 
+   * <p>This sets the {@link Closer} interface for this client.  Once set the client will call .onClose 
+   * on it once it a socket close is detected.</p> 
    * 
-   * @param closer sets this clients Closer interface.
+   * @param closer sets this clients {@link Closer} callback.
    */
-  protected void setCloser(Closer closer) {
-    if(! closed.get()) {
-      this.closer = closer;
-    }
-  }
+  public void setCloser(Closer closer);
   
   /**
-   * This is used to notify the clients Reader that a read has been added
-   */
-  protected void callReader() {
-    if(! closed.get() && sei != null && reader != null) {
-      sei.execute(new Runnable() {
-        @Override
-        public void run() {
-          getReader().onRead(Client.this);
-        }});
-    }
-  }
-  
-  /**
-   * Returns the currently set Reader callback.
+   * <p>Returns the currently set Reader callback. </p>
    * 
-   * @return the current Reader for this client.
+   * @return the current {@link Reader} callback for this client.
    */
-  protected Reader getReader() {
-    return reader;
-  }
+  public Reader getReader();
 
   /**
-   * This sets the Reader for the client. 
-   * The Reader is called every time a new read is added to the client
-   * Each time a Reader is called it should call .getRead() *once* on the client
+   * <p>This sets the Reader for the client.This should be set before adding the Client to the {@link SocketExecuterInterface}, 
+   * if its not there is a chance to miss data coming in on the socket.  Once set data received on the socket will be callback 
+   * on this Reader to be processed.</p>
    * 
-   * @param reader the Reader interface to set for this client.
+   * @param reader the {@link Reader} callback to set for this client.
    */
-  protected void setReader(Reader reader) {
-    if(! closed.get()) {
-      this.reader = reader;
-    }
-  }
+  public void setReader(Reader reader);
 
   /**
-   * This allows you to set/change the max buffer size for this client object.
-   * This is the in java memory buffer not the additional socket buffer the OS might setup. 
+   * <p>This allows you to set/change the max buffer size for this client object.
+   * This is the in java memory buffer not the additional socket buffer the OS might setup.</p>
    * 
-   * @param size in bytes.
+   * <p>In general this should be set to the max size you can deal with.  The lower this is the more often we
+   * will end up adding/removing the client from the selectors.  Both the read and write buffers are separate so 
+   * a client can use up to 2x the buffer size set here.  You need room in your heap for buffers for all clients.
+   * These buffers are not kept at full size, so clients will rarely use that much memory if the protocol parsing 
+   * and network are keeping up with the data going in/out.</p>
```

Comment:
In some ways this is dangerous.  For example you may have this set so you have enough heap/memory during normal conditions.  Something else on the system that causes the CPU to be constrained, causing the reading to become slow, or something happens to cause the consuming of the writes to be slow.

Now this condition is worse because we find we did not have enough heap for all the clients, and end up running out of memory too.

Not sure what, if anything, could be done about that...other than always having their full buffers allocated.